### PR TITLE
Override PlaceholderDimensions equality operator to avoid unnecessary TextPainter re-layouts

### PR DESCRIPTION
**Pull request by tgucio**
This PR overrides the equality operator in `PlaceholderDimensions`. Without this, each time `setPlaceholderDimensions()` is called with different `PlaceholderDimensions` instances carrying identical information (which will happen e.g. on resizing a `RenderEditable` with some `WidgetSpan`s), earlier (and still valid) layout information will be invalidated by calling `markNeedsLayout()`.

```
  void setPlaceholderDimensions(List<PlaceholderDimensions>? value) {
    if (value == null || value.isEmpty || listEquals(value, _placeholderDimensions)) {
      return;
    }

   [...]

    markNeedsLayout();
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by tgucio** [comment]
Looks like the failing tests are unrelated:
```
[36mRunning for google_maps_flutter_web...[0m
Running command: "flutter pub get" in /b/s/w/ir/x/t/flutter_plugins.EUFLEI/packages/google_maps_flutter/google_maps_flutter_web
Running "flutter pub get" in google_maps_flutter_web...          2,037ms
Running "flutter pub get" in example...                          2,176ms
Running command: "dart analyze --fatal-infos" in /b/s/w/ir/x/t/flutter_plugins.EUFLEI/packages/google_maps_flutter/google_maps_flutter_web
Analyzing google_maps_flutter_web...

   info - example/integration_test/google_maps_controller_test.dart:24:31 - 'returnNullOnMissingStub' is deprecated and shouldn't be used. Specify "missing stub" behavior with the [onMissingStub] parameter.. Try replacing the use of the deprecated member with the replacement. - deprecated_member_use
   info - example/integration_test/google_maps_controller_test.dart:25:32 - 'returnNullOnMissingStub' is deprecated and shouldn't be used. Specify "missing stub" behavior with the [onMissingStub] parameter.. Try replacing the use of the deprecated member with the replacement. - deprecated_member_use
   info - example/integration_test/google_maps_controller_test.dart:26:33 - 'returnNullOnMissingStub' is deprecated and shouldn't be used. Specify "missing stub" behavior with the [onMissingStub] parameter.. Try replacing the use of the deprecated member with the replacement. - deprecated_member_use
   info - example/integration_test/google_maps_controller_test.dart:27:31 - 'returnNullOnMissingStub' is deprecated and shouldn't be used. Specify "missing stub" behavior with the [onMissingStub] parameter.. Try replacing the use of the deprecated member with the replacement. - deprecated_member_use
   info - example/integration_test/google_maps_plugin_test.dart:20:33 - 'returnNullOnMissingStub' is deprecated and shouldn't be used. Specify "missing stub" behavior with the [onMissingStub] parameter.. Try replacing the use of the deprecated member with the replacement. - deprecated_member_use
```

**Review by LongCatIsLooong — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/painting/text_painter.dart [hunk 1]
```diff
@@ -108,6 +108,18 @@ class PlaceholderDimensions {
   ///  * [ui.PlaceholderAlignment.middle]
   final TextBaseline? baseline;
 
+  @override
+  bool operator ==(Object other) {
+    return other is PlaceholderDimensions
```
[the remark this answers is not in the document]
Comment (tgucio) [reply packages/flutter/lib/src/painting/text_painter.dart:116]: Good call - done.

**Comment by tgucio** [comment]
@LongCatIsLooong Could you slap the autosubmit label on the PR? I still need a find a sponsor for repo write access :).

**Review by yaakovschectman — APPROVED** [review]

**Pull request merged**